ex/loqui/lib/loqui/client.ex

```diff
@@ -0,0 +1,325 @@
+defmodule Loqui.Client do
+  defmodule State do
+    alias Loqui.Protocol.{
+      Codec,
+      Codecs,
+      Compressor,
+      Compressors
+    }
+
+    @type t :: %__MODULE__{
+      port: pos_integer,
+      host: char_list,
+      tcp_opts: Loqui.Client.tcp_opts,
+      recv_timeout: pos_integer,
+      loqui_path: String.t,
+      buffer: binary,
+      sequence: Loqui.Client.sequence,
+      ping_interval: pos_integer,
+      registered_compressors: %{String.t => Compressor.t},
+      registered_codecs: %{String.t => Codec.t},
+      compressor: Compressor.t,
+      codec: Codec.t,
+      waiters: %{Loqui.Client.sequence => pid}
+    }
+
+    defstruct host: nil,
+      port: nil,
+      sock: nil,
+      tcp_opts: nil,
+      connect_timeout: nil,
+      recv_timeout: nil,
+      loqui_path: nil,
+      buffer: <<>>,
+      sequence: 1,
+      ping_interval: nil,
+      registered_compressors: %{},
+      registered_codecs: %{},
+      compressor: Compressors.NoOp,
+      codec: Codecs.Erlpack,
+      waiters: %{}
+  end
+
+  alias Loqui.Protocol.{Codec, Codecs, Compressor, Compressors, Frames, Parser}
+  use Connection
+  use Loqui.{Opcodes, Types}
+  require Logger
+
+  @type sequence :: pos_integer
```

Comment:
This could be more accurately expressed as `1..unquote(@max_sequence)` if attribute registered before the type